### DISCUSSION TITLE
Bottom Sheet Buttons

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1246] [Proposals]
### Goal

Novo aplicativo Vivo -  Proposta Work & Co aprovada pelo time de Design e Ops da Vivo.

### Context

Pensamos em usar o Bottom Sheet para esse tipo de situação porque usar o Dialog Native parece trazer uma ruptura na experiência, e sugere uma criticidade muito grande para a ação do usuário.
 
Nossa proposta é de padronizar a experiência na navegação com Bottom sheet para esses casos. 

A experiência parece ficar mais mais fluida.

### Impact

Experiência fluida da pessoa usuária.

### Discussion Questions

É possível utilizar esse componente nos CMS?

### Additional considerations

Link task Jira: https://jira.tid.es/browse/OBVIVO-1825

### Do you have a visual proposal?

https://www.figma.com/file/9ocLzB36RBaApOdPweYTKX/Bottom-Sheet-Buttons?type=design&node-id=0%3A1&t=DzjOPn0tovZ0bJAQ-1

**Comment by AnaMontes11** [comment]
Hola @annefranca, hemos evaluado vuestra propuesta y consideramos que como proponéis, tiene sentido incluirla para casos en los que se busca una experiencia más fluida.

Nos surgen algunas dudas que nos gustaría aclarar con vosotros sobre cómo vemos que tiene sentido integrar esta tipología de bottom sheet dado que tendríamos 3 componentes muy cercanos a cubrir la misma necesidad:

1. Dialog Confirm (native)
2. Action Sheet (native)
3. Bottom Sheet

Nos preocupa que no quede clara la diferencia de uso entre ellos para los diseñadores. 

Nuestro planteamiento es el siguiente:

**Dialog Confirm:** se utiliza para obtener la confirmación del usuario antes de realizar una acción importante o irreversible, bloqueando al usuario. Se muestra en el centro de la pantalla y generalmente contiene un mensaje y dos opciones (como "Aceptar" y "Cancelar") para que el usuario elija. Se utiliza cuando se necesita una confirmación explícita del usuario antes de realizar una acción crítica, como eliminar un elemento importante o realizar una operación que no se pueda deshacer fácilmente.

**Action Sheet:** se despliega desde la parte inferior de la pantalla y muestra varias acciones relacionadas con el contenido de la vista actual. Las acciones se muestran en forma de lista para representar cada acción. Dependiendo de la implementación, la acción seleccionada puede activarse inmediatamente o mostrar una pantalla secundaria relacionada con esa opción. (deberíamos cargarnos la action sheet?)

**Bottom Sheet:**  se utiliza generalmente para mostrar opciones o acciones relacionadas con el contenido actual en la pantalla y se desea que el usuario interactúe con ellas de forma rápida sin cambiar de pantalla. La tipología de buttons para este caso NO plantea una decisión al usuario, sino una bifurcación de la navegación. 

Por ejemplo, el caso que más se parecería a esto sería el "Terminar compra" y "seguir comprando"

Lo importante es que quede suficientemente claras las diferencias y se entiende el contexto de uso para que no existan solapamientos entre uno u otro componente.

Os encaja esta definición? Ya que podría ser posible usar ambos componentes en diferentes partes de la aplicación, según corresponda a un caso de uso u otro, queremos que para que coexistan, su uso debe estar claro y bien diferenciado.

**Comment by BrunoBiagioniNeto** [comment]
Tal y como se acordó en la última reunión con la selección española.

**Dialog** - Decisión sistémica (eliminar, eliminar, confirmar editar)

**Bottom sheet** - Es una interacción del usuario con la pantalla con un comportamiento de bloqueo para ayudar a los usuarios a concentrarse en las opciones disponibles.

**Bottom sheet  - Plantilla 1 - Action Sheet** (actual - selección de opciones disponibles, generalmente una lista)
**Bottom sheet  - Plantilla 2 - Decision Making** (nuevo - toma de decisiones empresariales)

*Se acordó que la ActionSheet es un ítem dentro de Bottom sheet.

**Comment by yceballost** [comment]
Cierro y lo metemos como tarea a priorizar
